### PR TITLE
feature(producerImpl): use interruptible instead of delay

### PR DESCRIPTION
**Pull request by ERobertGII**
Compliment pr to this [one](https://github.com/Banno/kafka4s/pull/835)
change `producer` operations to use `interruptible `instead of `delay`.

[KAYAK-3388]

[KAYAK-3388]: https://banno-jha.atlassian.net/browse/KAYAK-3388?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: core/src/main/scala/com/banno/kafka/producer/ProducerImpl.scala [hunk 1]
```diff
@@ -33,22 +33,23 @@ case class ProducerImpl[F[_], K, V](p: Producer[K, V])(implicit
...
+  def close: F[Unit] = F.interruptible(p.close())
   def close(timeout: FiniteDuration): F[Unit] =
-    F.delay(p.close(java.time.Duration.ofMillis(timeout.toMillis)))
-  def commitTransaction: F[Unit] = F.delay(p.commitTransaction())
-  def flush: F[Unit] = F.delay(p.flush())
-  def initTransactions: F[Unit] = F.delay(p.initTransactions())
-  def metrics: F[Map[MetricName, Metric]] = F.delay(p.metrics().asScala.toMap)
+    F.interruptible(p.close(java.time.Duration.ofMillis(timeout.toMillis)))
+  def commitTransaction: F[Unit] = F.interruptible(p.commitTransaction())
+  def flush: F[Unit] = F.interruptible(p.flush())
```
Comment (ERobertGII) [inline core/src/main/scala/com/banno/kafka/producer/ProducerImpl.scala:42]: Is `flush `an example of something that could remain `.delay`?

File: core/src/main/scala/com/banno/kafka/producer/ProducerImpl.scala [hunk 2]
```diff
@@ -89,7 +90,7 @@ case class ProducerImpl[F[_], K, V](p: Producer[K, V])(implicit
         record,
         { (rm, e) => callback(Option(e).toLeft(rm).toTry) },
       )
-    ).map(jf => F.delay(jf.cancel(true)).void)
+    ).map(jf => F.interruptible(jf.cancel(true)).void)
```
Comment (zcox) [inline core/src/main/scala/com/banno/kafka/producer/ProducerImpl.scala:93]: Sorry, I should've mentioned: we've already gone through all of the send-related methods, and those delays can remain.

File: core/src/main/scala/com/banno/kafka/producer/ProducerImpl.scala [hunk 3]
```diff
@@ -33,22 +33,23 @@ case class ProducerImpl[F[_], K, V](p: Producer[K, V])(implicit
     F: Async[F],
     T: Trace[F],
 ) extends ProducerApi[F, K, V] {
-  def abortTransaction: F[Unit] = F.delay(p.abortTransaction())
-  def beginTransaction: F[Unit] = F.delay(p.beginTransaction())
-  def close: F[Unit] = F.delay(p.close())
+  def abortTransaction: F[Unit] = F.interruptible(p.abortTransaction())
```
Comment (ERobertGII) [inline core/src/main/scala/com/banno/kafka/producer/ProducerImpl.scala:36]: is `abortTransaction` able to stay `.delay`?

File: core/src/main/scala/com/banno/kafka/producer/ProducerImpl.scala (hunk 2 above)
Comment (ERobertGII) [reply core/src/main/scala/com/banno/kafka/producer/ProducerImpl.scala]: got it! ty.

**Review by zcox — APPROVED** [review]

File: core/src/main/scala/com/banno/kafka/producer/ProducerImpl.scala [hunk 4]
```diff
@@ -106,15 +107,16 @@ case class ProducerImpl[F[_], K, V](p: Producer[K, V])(implicit
...
+    F.interruptible(sendRaw(record))
       .void // discard the returned JFuture[RecordMetadata]
 
   /** Like `sendSync`, but blocks a compute thread awaiting the acknowledgement.
     * Do not use.
     */
   @deprecated("Use sendAsync, or send in kafka4s-6.x", "5.0.4")
   def sendSync(record: ProducerRecord[K, V]): F[RecordMetadata] =
-    F.delay(sendRaw(record)).flatMap(jFut => F.interruptible(jFut.get()))
+    F.delay(sendRaw(record))
```
Comment (rossabaker) [inline core/src/main/scala/com/banno/kafka/producer/ProducerImpl.scala:118]: Isn't this one interruptible?  Doesn't `sendRaw(record)` block if the internal buffer is full?

Comment (zcox) [reply core/src/main/scala/com/banno/kafka/producer/ProducerImpl.scala:118]: Ah yeah, Ross is right. We'll be getting rid of `sendSync` at some point, but should make it work better until then.

Comment (ERobertGII) [reply core/src/main/scala/com/banno/kafka/producer/ProducerImpl.scala:118]: copy that.  on it! ty.

**Review by zcox — APPROVED** [review]

**Pull request merged**